qp_klp/process_amplicon_job.py

```diff
@@ -0,0 +1,391 @@
+from metapool.prep import remove_qiita_id
+from os import listdir, makedirs
+# from os import walk
+# from os.path import basename, join, exists
+from os.path import exists, join, isfile, basename
+from qiita_client import ArtifactInfo
+# from qp_klp.klp_util import map_sample_names_to_tube_ids
+from random import choices
+from sequence_processing_pipeline.ConvertJob import ConvertJob
+from sequence_processing_pipeline.FastQCJob import FastQCJob
+# from sequence_processing_pipeline.GenPrepFileJob import GenPrepFileJob
+from sequence_processing_pipeline.Pipeline import Pipeline
+from sequence_processing_pipeline.PipelineError import PipelineError
+from subprocess import Popen, PIPE
+import pandas as pd
+import shutil
+
+
+def process_amplicon(mapping_file_path, qclient, run_identifier, out_dir,
```

Comment:
I'm actually surprised that the metagenomic and amplicon sections are so different that deserve their own full method - I would have thought that they were basically the same with a few if/else's, is that right?

Reply:
I think that from a working amplicon pipeline we could mechanically merge them back together. Developing the amplicon pipeline while preserving the robustness of the metagenomic pipeline got a little hairy so I elected to separate the two. I also tried to change the Job objects as little as possible since they would either need to be shared or duplicated as well. That seemed to work out in that ConvertJob required no change and QCJob could be dropped entirely. FastQCJob required just a small change to support amplicon. Then it became a matter of tweaking the inputs and outputs between each Job so that's where the differences come in I suppose. Small changes like copying files into the directory structure FastQCJob was expecting were needed and not being able to generate failed-samples records became a thing.

Reply:
Mapping file functionality is currently a part of Pipeline and if the mapping file is really the analog of a sample-sheet, then the functionality would be better off in the metapool package.  The copying of files into QCJob's output file/directory structure could also be considered a degenerate case of QCJob operation so that could be refactored into QCJob as well. Those two items and possibly refactoring both lines into one or standardizing for potentially new pipelines in the future would be what comes to mind.